### PR TITLE
Use GitHub cache instead of FTP to use OpenDDS artifacts within pull request and Release workflows

### LINKED ISSUE #42: Use GitHub cache instead of FTP site to use OpenDDS artifacts within Pull Request and Release workflows
Forced "squash and merge" due to space limit reached.

_Originally posted by @masesdevelopers in https://github.com/masesgroup/DataDistributionManager/issues/41#issuecomment-1673181541_

### PR DESCRIPTION
**Pull request by masesdevelopers**
## Description

The PR updates workflows to use GitHub cache to store OpenDDS artifacts within Pull Requests and Release.
This speedup both actions because of stored information.
If cache is not available it will be rebuilt before execute the action

## Related Issue

Closed #42 

## How Has This Been Tested?

Tested running workflows

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -20,18 +20,27 @@ jobs:
...
+          lookup-only: true
+          enableCrossOsArchive: true
+          path: ./OutputOpenDDS/
+          key: OpenDDS_3.25
+          
+     # - name: Exists OpenDDS
+     #   id: existsOpenDDS
+     #   continue-on-error: true
+     #   run: |
+     #     python $GITHUB_WORKSPACE/third_party/CommonTools/scripts/utilsftp.py -c check -s ${{ secrets.FTP_STORE_SITE }} -u ${{ secrets.FTP_STORE_USER }} -p ${{ secrets.FTP_STORE_PASSWORD }} -r OpenDDS_Build_3.25.zip -l $GITHUB_WORKSPACE/OpenDDS_Build_3.25.zip
```
Comment (mariomastrodicasa) [inline .github/workflows/build.yaml:38]: Clean code

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -254,27 +273,35 @@ jobs:
...
+    #      python %GITHUB_WORKSPACE%\third_party\CommonTools\scripts\utilsftp.py -c download -s ${{ secrets.FTP_STORE_SITE }} -u ${{ secrets.FTP_STORE_USER }} -p ${{ secrets.FTP_STORE_PASSWORD }} -r OpenDDS_Build_3.25.zip -l %GITHUB_WORKSPACE%\OpenDDS_Build_3.25.zip
+    #    env:
+    #      GITHUB_WORKSPACE: $GITHUB_WORKSPACE
+    #      
+    #  - name: Download OpenDDS Headers
+    #    shell: cmd
+    #    run: |
+    #      python %GITHUB_WORKSPACE%\third_party\CommonTools\scripts\utilsftp.py -c download -s ${{ secrets.FTP_STORE_SITE }} -u ${{ secrets.FTP_STORE_USER }} -p ${{ secrets.FTP_STORE_PASSWORD }} -r OpenDDS_Headers_3.25.zip -l %GITHUB_WORKSPACE%\OpenDDS_Headers_3.25.zip
+    #    env:
+    #      GITHUB_WORKSPACE: $GITHUB_WORKSPACE
```
Comment (mariomastrodicasa) [inline .github/workflows/build.yaml:296]: Clean code

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -86,49 +227,73 @@ jobs:
...
+    #      xcopy .\lib\*.dll ..\..\Output\x86\
+    #      xcopy .\lib\*.pdb ..\..\Output\pdb32\
+    #      xcopy .\lib\*.lib ..\..\Output\lib32\
+    #      xcopy .\bin\*.exe ..\..\Output\bin32\
+    #      xcopy .\bin\*.pdb ..\..\Output\pdb32\
+    #      xcopy .\ACE_wrappers\lib\*.dll ..\..\Output\x86\
+    #      xcopy .\ACE_wrappers\lib\*.pdb ..\..\Output\pdb32\
+    #      xcopy .\ACE_wrappers\lib\*.lib ..\..\Output\lib32\
+    #      xcopy .\ACE_wrappers\bin\*.exe ..\..\Output\bin32\
+    #      xcopy .\ACE_wrappers\bin\*.pdb ..\..\Output\pdb32\
```
Comment (mariomastrodicasa) [inline .github/workflows/release.yaml:296]: Clean code

**Review by mariomastrodicasa — COMMENTED** [review]
Clean up commented code

File: .github/workflows/build.yaml (hunk 2 above)
Comment (masesdevelopers) [reply .github/workflows/build.yaml:296]: Leaved in workflow till completed at least a first release cycle

File: .github/workflows/build.yaml (hunk 1 above)
Comment (masesdevelopers) [reply .github/workflows/build.yaml:38]: Leaved in workflow till completed at least a first release cycle

File: .github/workflows/release.yaml (hunk 3 above)
Comment (masesdevelopers) [reply .github/workflows/release.yaml:296]: Leaved in workflow till completed at least a first release cycle

**Pull request merged**